src/main/java/frc/robot/RobotContainer.java

```diff
@@ -98,33 +89,32 @@
   private final ClimberSubsystem climberSubsystem = new ClimberSubsystem();
   private final ShooterHoodSubsystem shooterHoodSubsystem = new ShooterHoodSubsystem();
   private final Limelight limelight = new Limelight(limelightTab, matchTab, 0, 0);
-  private final ShooterSubsystem shooterSubsystem = new ShooterSubsystem(shooterHoodSubsystem);
+  private final ShooterSubsystem shooterSubsystem = new ShooterSubsystem(shooterHoodSubsystem, matchTab);
   private final StorageSubsystem storageSubsystem = new StorageSubsystem(powerDistributionPanel);
   private final LEDSubsystem ledSubsystem = new LEDSubsystem();
 
   // ---------------------------------------Commands--------------------------------------------
   private final AutonDriveCommand autonDriveCommand = new AutonDriveCommand(falconDriveSubsystem, limelight);
-  private final ShootFromWall shootFromWall = new ShootFromWall(falconDriveSubsystem, shooterSubsystem,
-      storageSubsystem);
-  // private final Command autonomousCommand = new
-  // ParallelCommandGroup(autonDriveCommand,
-  // new AutoStowClimberCommand(climberSubsystem));
-  private final DriveToWall driveToWall = new DriveToWall(falconDriveSubsystem, shooterSubsystem, storageSubsystem);
 
   private final DriveCommand driveCommand = new DriveCommand(falconDriveSubsystem, mainController, driveConditioner);
 
   private final Command shootCommand = new RunCommand(shooterHoodSubsystem::deployHood, shooterHoodSubsystem)
       .alongWith(new RunCommand(shooterSubsystem::shoot, shooterSubsystem))
       .alongWith(new RunCommand(limelight::lightsOn, limelight));
 
-  private final Command shootIdleAutoCommand = new RunCommand(shooterSubsystem::idle, shooterSubsystem)
-      .alongWith(new RunCommand(shooterHoodSubsystem::stowHood, shooterHoodSubsystem));
+  private final Command shootFromLine = new ParallelCommandGroup(
+      new RunCommand(shooterSubsystem::shoot, shooterSubsystem).withTimeout(10),
+      new RunCommand(shooterHoodSubsystem::deployHood, shooterHoodSubsystem).withTimeout(10),
+      new RunCommand(intakeSubsystem::runIntakeFwd, intakeSubsystem).withTimeout(10)
+          .andThen(new RunCommand(() -> falconDriveSubsystem.drive(0.5, 0), falconDriveSubsystem)));
 
-  private final ShootFromLine shootFromLine = new ShootFromLine(falconDriveSubsystem, shooterSubsystem,
-      storageSubsystem, intakeSubsystem, shooterHoodSubsystem);
+  private final Command shootFromWall = new ParallelCommandGroup(
+      new RunCommand(() -> falconDriveSubsystem.drive(0.5, 0)).withTimeout(5).andThen(
+          new RunCommand(shooterSubsystem::shoot, shooterSubsystem),
+          new RunCommand(() -> shooterHoodSubsystem.setHoodPosition(60), shooterHoodSubsystem),
```

Comment:
setHoodPosition(magic_number).

I still don't understand why the shooterHood is treated differently from the shooter.  There should be a hierarchy of subsystems such that if you want to shoot, you send a command to the shooter and it figures out what to do with the hood to make the shots correctly.  The top level command shouldn't need to know about both.